File: data_types.py
from pydantic import BaseModel, Field


class EducationItem(BaseModel):
    institution: str
    degree: str
    duration: str
    marks: str


class CourseItem(BaseModel):
    name: str
    platform: str
    certificate: str | None
    year: str


class EducationData(BaseModel):
    education: list[EducationItem]
    courses: list[CourseItem]


class ExperienceDuration(BaseModel):
    start: str
    end: str


class ExperienceItem(BaseModel):
    company: str
    role: str
    duration: ExperienceDuration
    responsibilities: list[str]


ExperienceData = list[ExperienceItem]


class OpenSourceItem(BaseModel):
    name: str
    description: str
    link: str


OpenSourceData = list[OpenSourceItem]


class ContactInformation(BaseModel):
    Email: str
    Phone_number: str | None = Field(None, alias="Phone number")


class SocialMediaProfiles(BaseModel):
    LinkedIn: str
    GitHub: str
    Twitter: str | None = None


class PersonalInformation(BaseModel):
    Name: str
    DOB: str
    Residing_location: str = Field(alias="Residing location")
    Nationality: str
    Languages: list[str]
    Hobbies: list[str] | None = None
    family: dict | None = None
    Contact_information: ContactInformation = Field(alias="Contact information")
    Social_media_profiles: SocialMediaProfiles = Field(alias="Social media profiles")
    Professional_summary: str = Field(alias="Professional summary")
    Skills_and_competencies: list[str] = Field(alias="Skills and competencies")


class Publication(BaseModel):
    description: str
    link: str


class PersonalData(BaseModel):
    description: str
    personal_information: PersonalInformation
    publications: Publication


class ProjectItem(BaseModel):
    project_id: str
    project_name: str
    description: str
    key_points: list[str]
    tech_stack: list[str]
    tags: list[str]
    organization: str
    duration: str


ProjectsData = list[ProjectItem]


class BlogItem(BaseModel):
    headline: str
    description: str
    link: str


BlogsData = list[BlogItem]

File: dataTypes.ts
export interface EducationItem {
    institution: string;
    degree: string;
    duration: string;
    marks: string;
}

export interface CourseItem {
    name: string;
    platform: string;
    certificate: string | null;
    year: string;
}

export interface EducationData {
    education: EducationItem[];
    courses: CourseItem[];
}

export interface ExperienceItem {
    company: string;
    role: string;
    duration: {
        start: string;
        end: string;
    };
    responsibilities: string[];
}

export type ExperienceData = ExperienceItem[];

export interface OpenSourceItem {
    name: string;
    description: string;
    link: string;
}

export type OpenSourceData = OpenSourceItem[];

export interface PersonalInformation {
    Name: string;
    DOB: string;
    "Residing location": string;
    Nationality: string;
    Languages: string[];
    Hobbies: string[] | null;
    family: any | null;
    "Contact information": {
        Email: string;
        "Phone number": string | null;
    };
    "Social media profiles": {
        LinkedIn: string;
        GitHub: string;
        Twitter: string | null;
    };
    "Professional summary": string;
    "Skills and competencies": string[];
}

export interface Publication {
    description: string;
    link: string;
}

export interface PersonalData {
    description: string;
    personal_information: PersonalInformation;
    publications: Publication;
}

export interface ProjectItem {
    project_id: string;
    project_name: string;
    description: string;
    key_points: string[];
    tech_stack: string[];
    tags: string[];
    organization: string;
    duration: string;
}

export type ProjectsData = ProjectItem[];
